### DISCUSSION TITLE
Option to auto-play?

### DISCUSSION THREAD

**Discussion by ibillwilson** [ruffle-rs/ruffle#839] [General] [answered]
Is there an API option to auto-play when the swf is ready? I don't see anything in the documentation or the code.

**Comment by Herschel** [comment]
Not yet, but it's definitely something we will add!
(The one gotcha is that the browser often prevents audio from auto-playing; so audio may not play until a user interacts with the movie).

**Comment by ousia** [comment, marked as answer]
@ibillwilson,

this is the code you need:

``` html
<script src="./ruffle/ruffle.js"></script>

<script>
window.RufflePlayer.config = {
    autoplay: "auto"
};
</script>
```

[Edited after realizing my previous answer was wrong.]

**Reply by n0samu** [reply]
The correct value is `autoplay: "on"`, if you want the Flash to unconditionally autoplay.

**Comment by ousia** [comment]
@ibillwilson,

BTW, `autoplay` is now the default value.

Best wishes for 2021 🎉!

**Reply by ibillwilson** [reply]
Thank you!

**Comment by BinBashBanana** [comment]
@ousia I could not get your solution to work. Here is my code, did I do anything wrong?
```html
<!DOCTYPE html>
<html>
	<head>
		<script src="./ruffle.js"></script>
		<script>
			window.addEventListener("load", function(e) {
				window.RufflePlayer = window.RufflePlayer || {};
				window.RufflePlayer.config = {autoplay: "auto"};
				const ruffle = window.RufflePlayer.newest();
				var player = ruffle.createPlayer();
				player.classList.add("gembed");
				document.body.appendChild(player);
				player.load({url: "./bulletbill.swf"});
			}, false);
		</script>
	</head>
	<body>
	</body>
</html>
```

**Reply by ousia** [reply]
@BinBashBanana, JS is almost Greek to me.

But if the auto-play is the default behavior, I would remove `window.RufflePlayer.config = {autoplay: "auto"};`.

I don’t know whether Chrome has issues with auto-play. But I’m only using Firefox.

If you don’t get auto-play after removing the mentioned line (at least, with Firefox), please open an issue.

**Reply by BinBashBanana** [reply]
It appears to work on firefox just fine, but chrome prevents it. Oh well...
